### PR TITLE
Run e2e tests with PHAR additionally on PHP 8.1

### PR DESCRIPTION
**Pull request by maks-rafalko**
Trying to catch a bug when prefixed PHAR fails with fatal error

```bash
PHP Fatal error:  Uncaught TypeError: Composer\XdebugHandler\XdebugHandler::setLogger(): Argument #1 ($logger) must be of type Psr\Log\LoggerInterface, _HumbugBox4beab5586524\Infection\Logger\ConsoleLogger given, called in phar:///homeinfection/build/infection.phar/src/Console/XdebugHandler.php on line 15 and defined in /infection/vendor/composer/xdebug-handler/src/XdebugHandler.php:93
Stack trace:
```

### CODE REVIEW

**Comment by maks-rafalko** [comment]
Box does not work with PHP 8.1 and Infection:

```
 PHP Fatal error:  Uncaught Error: Class "_HumbugBox72d5912ce99f\Symfony\Component\Console\Application" not found in phar:///home/runner/work/infection/infection/build/infection.phar/src/Console/Application.php:20
Stack trace:
#0 phar:///home/runner/work/infection/infection/build/infection.phar/.box/vendor/composer/ClassLoader.php(478): include()
#1 phar:///home/runner/work/infection/infection/build/infection.phar/.box/vendor/composer/ClassLoader.php(346): Composer\Autoload\includeFile()
#2 phar:///home/runner/work/infection/infection/build/infection.phar/bin/infection(71): Composer\Autoload\ClassLoader->loadClass()
#3 /home/runner/work/infection/infection/build/infection.phar(14): require('...')
#4 {main}
  thrown in phar:///home/runner/work/infection/infection/build/infection.phar/src/Console/Application.php on line 20
make: *** No rule to make target 'test-e2e'.  Stop.
```

**Comment by maks-rafalko** [comment]
Finally, compiling PHAR works on PHP 8.1 and now is tested on our matrix.

Next steps: make it working with Composer 2.2 (now composer version is fixed at 2.1 on GHA).

@theofidry if you find a minute, would be nice to check `scoper.inc.php` changes as I'm barely have an idea what I've done there

File: scoper.inc.php [hunk 1]
```diff
@@ -33,13 +33,6 @@
 
 declare(strict_types=1);
 
-use Isolated\Symfony\Component\Finder\Finder;
```
Comment (theofidry) [inline scoper.inc.php:36]: Was this necessary? This means the Symfony polyfills are no longer excluded.

I think the canonical way to do this slightly changed though: https://github.com/humbug/php-scoper/blob/master/docs/further-reading.md#polyfills

Comment (maks-rafalko) [reply scoper.inc.php:36]: Given we have green builds, do we really need to exclude symfony polyfills?

**Pull request merged**